rtrlib/lib/vector.c

```diff
@@ -0,0 +1,154 @@
+
+
+#include "vector.h"
+
+#include <math.h>
+
+int aspa_vector_create(struct aspa_vector **vector_pointer)
+{
+	const size_t DEFAULT_INITIAL_SIZE = 128;
+
+	// allocation the chunk of memory
+	struct aspas_record *data_field =
+		(struct aspas_record *)malloc(sizeof(struct aspas_record) * DEFAULT_INITIAL_SIZE);
+
+	// malloc failed so returning an error
+	if (data_field == NULL) {
+		return -1;
+	}
+
+	struct aspa_vector *vector = (struct aspa_vector *)malloc(sizeof(struct aspa_vector));
+
+	if (vector == NULL) {
+		return -1;
+	}
+
```

Comment:
Isn't

```c
if (!vector) {
 return -1;
}
```
enough?

Reply:
yes it is.  -> but this I think this verbosity is good here.